test/service_binding_test/service_binding_test.go

```diff
@@ -675,6 +677,115 @@ var _ = DescribeTestsFor(TestCase{
 								})
 
 								if testCase.async {
+									When("maximum polling duration is reached while polling", func() {
```

Comment:
do we cover both cases with tetsts currently ?
a case testing a situation in which the action timeout is smaller than polling timeout and verifying that we actually eventually fail the op due to action timeout reached and not due to action timeout. 
And also vise versa action timeout > polling timeout => fail due to polling timeout
If yes there need to be proper BDD context separation indicating that, if not let's add the missing one and also bdd context  separation